src/transport/xqc_frame.c

```diff
@@ -2026,6 +2041,53 @@ xqc_process_max_path_id_frame(xqc_connection_t *conn, xqc_packet_in_t *packet_in
     return ret;
 }
 
+
+xqc_int_t
+xqc_process_path_blocked_frame(xqc_connection_t *conn, xqc_packet_in_t *packet_in)
+{
+    xqc_int_t ret = XQC_ERROR;
+    uint64_t max_path_id, new_max_path_id;
+
+    ret = xqc_parse_path_blocked_frame(packet_in, &max_path_id);
+    if (ret != XQC_OK) {
+        xqc_log(conn->log, XQC_LOG_ERROR,
+                "|xqc_process_max_paths_frame error|");
+        return ret;
+    }
+
+    xqc_log(conn->log, XQC_LOG_DEBUG,
+            "|max_path_id:%ui|pre_local_max_path_id:%ui|", max_path_id, conn->local_max_path_id);
+
+    if (conn->local_max_path_id < max_path_id) {
+        xqc_log(conn->log, XQC_LOG_ERROR,
+                "|invalid path blocked frame|");
+        return -XQC_EIGNORE_PKT;
+    }
+
+    if (conn->local_max_path_id > max_path_id) {
+        xqc_log(conn->log, XQC_LOG_DEBUG,
+                "|received out-dated path blocked frame|local_max_path_id:%ui|max_path_id:%ui|", conn->local_max_path_id, max_path_id);
+        return XQC_OK;
+    }
+
+    uint64_t pre_max_path_id = conn->local_max_path_id;
+    conn->local_max_path_id += (conn->local_max_path_id + 1) / 2;
```

Comment:
We should add some limitations here. For example, it should refuse to increase this value once some thresholds are reached.